File: apps/macos/Mollotov/Handlers/HistoryHandler.swift
import Foundation

/// API handler for navigation history: list, clear.
struct HistoryHandler {
    let context: HandlerContext

    func register(on router: Router) {
        router.register("history-list") { body in await list(body) }
        router.register("history-clear") { _ in await clear() }
    }

    @MainActor
    private func list(_ body: [String: Any]) async -> [String: Any] {
        let limit = body["limit"] as? Int ?? 100
        let entries = HistoryStore.shared.toJSON()
        let sliced = Array(entries.prefix(limit))
        return successResponse(["entries": sliced, "total": entries.count])
    }

    @MainActor
    private func clear() async -> [String: Any] {
        HistoryStore.shared.clear()
        return successResponse(["cleared": true])
    }
}

File: apps/android/app/src/main/java/com/kelpie/browser/handlers/HistoryHandler.kt
package com.kelpie.browser.handlers

import com.kelpie.browser.browser.HistoryStore
import com.kelpie.browser.network.Router
import com.kelpie.browser.network.successResponse

class HistoryHandler(
    private val ctx: HandlerContext,
) {
    fun register(router: Router) {
        router.register("history-list") { list(it) }
        router.register("history-clear") { clear() }
    }

    private suspend fun list(body: Map<String, Any?>): Map<String, Any?> {
        val limit = (body["limit"] as? Number)?.toInt() ?: 100
        val entries = HistoryStore.toJSON()
        return successResponse(mapOf("entries" to entries.take(limit), "total" to entries.size))
    }

    private suspend fun clear(): Map<String, Any?> {
        HistoryStore.clear()
        return successResponse(mapOf("cleared" to true))
    }
}
